Dockerfile with updated installation of vdr-tools

Abdulbois opened the pull request:
Remove evernym related parts
Add needed deps for vdr-tools
Enable building vdr-tools from bhutan-ndi/verity repo

Abdulbois committed:
Remove evernym related parts
Add needed deps for vdr-tools
Enable building vdr-tools from bhutan-ndi/verity repo

andkononykhin commented on verity/src/docker/Dockerfile:
```diff
@@ -1,41 +1,23 @@
+        software-properties-common \
+        libzmq3-dev \
+        libsodium23 \
+        libssl1.1
```
please add a cleanup command here ([ref](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#apt-get)) to remove apt-get's cache and reduce the layer's size

andkononykhin commented on verity/src/docker/Dockerfile:
```diff
@@ -1,41 +1,23 @@
-    sed -i '/repo\.corp\.evernym\.com/d' /etc/apt/sources.list && \
-    rm -rf /var/lib/apt/lists/* \
-    ; exit 0
+# Install vdr-tools from the debian package
```
please make theme configurable, e.g.
```
ARG VERITY_VERSION
ENV VERITY_VERSION=${VERITY_VERSION:-2.19.0.0}
ARG VDRTOOLS_VERSION
ENV VDRTOOLS_VERSION=${VDRTOOLS_VERSION:-0.8.6}
RUN curl -L https://github.com/Bhutan-NDI/verity/releases/download/v${VERITY_VERSION}/libvdrtools_${VDRTOOLS_VERSION}-bionic_amd64.deb
```

andkononykhin commented on verity/src/docker/Dockerfile:
```diff
@@ -1,41 +1,23 @@
-    rm -rf /var/lib/apt/lists/* \
-    ; exit 0
+# Install vdr-tools from the debian package
+RUN curl -L https://github.com/Bhutan-NDI/verity/releases/download/v2.19.0.0/libvdrtools_0.8.6-bionic_amd64.deb --output libvdrtools.deb && \
```
why the version has 4 numbers?

Abdulbois replied:
I took version from verity-codebase. Seems they are used 4 numbers convention to publish releases

andkononykhin commented on verity/src/docker/Dockerfile:
```diff
@@ -1,41 +1,23 @@
-    ; exit 0
+# Install vdr-tools from the debian package
+RUN curl -L https://github.com/Bhutan-NDI/verity/releases/download/v2.19.0.0/libvdrtools_0.8.6-bionic_amd64.deb --output libvdrtools.deb && \
+    dpkg -i libvdrtools.deb && rm libvdrtools.deb
```
split into two lines please

Abdulbois committed:
Optimize dockerfile

Add/configure VERITY_VERSION and VDRTOOLS_VERSION args
Add cleanup command to remove apt-get's cache
Split installing vdr-tools into two lines

andkononykhin commented on verity/src/docker/Dockerfile:
```diff
@@ -10,32 +13,20 @@ RUN apt-get update && \
-    rm -rf /var/lib/apt/lists/* \
-    ; exit 0
+# Install the vdr-tools then remove debian package
+RUN dpkg -i libvdrtools.deb && rm libvdrtools.deb
```
please run that at the same RUN statement (as it was before)

andkononykhin commented on verity/src/docker/Dockerfile:
```diff
@@ -1,7 +1,10 @@
+ENV VERITY_VERSION=${VERITY_VERSION:-2.19.0.0}
+
+ARG VDRTOOLS_VERSION
+ENV VDRTOOLS_VERSION=${VDRTOOLS_VERSION:-0.8.6}
```
please move all the args definitions right before where it is used (before line 26)

andkononykhin reviewed (commented):
@Abdulbois please check my comments

Abdulbois committed:
Optimize

Move ARGs near to used places
Make installation of vdr-tools in one RUN command
